### PR TITLE
[Submission] Warmdown Scheduling - 1.2430 BPB on 8×H100 SXM

### PR DESCRIPTION
**Pull request by MajdiZamim**
## Submission: Optimal LR Warmdown Scheduling

**Score:** 1.2430 BPB  
**Hardware:** 8×H100 SXM, 10 minutes  
**Author:** Majdi Zamim (@MajdiZamim)

## Key Insight

The default `WARMDOWN_ITERS=1200` never fires when training is 
wallclock-capped at 600 seconds on slower hardware, causing the 
model to terminate without proper LR decay.

By setting `WARMDOWN_ITERS=3000`, the cosine warmdown fires in 
the final ~30% of training steps, producing better convergence.

## Command
```bash
NCCL_IB_DISABLE=1 \
RUN_ID=submission_final \
VOCAB_SIZE=1024 \
NUM_LAYERS=9 \
MODEL_DIM=512 \
NUM_HEADS=8 \
NUM_KV_HEADS=4 \
TRAIN_BATCH_TOKENS=524288 \
MAX_WALLCLOCK_SECONDS=600 \
VAL_LOSS_EVERY=200 \
WARMDOWN_ITERS=3000 \
torchrun --standalone --nproc_per_node=8 train_gpt.py
```

## Files
- `README.md` — approach description
- `submission.json` — metadata
- `train.log` — full training log
- `train_gpt.py` — training script

### CODE REVIEW

**Comment by MajdiZamim** [comment]
Updated submission with improved score: 1.1708 BPB (previous: 1.2430 BPB)

Key changes:
- NCCL_IB_DISABLE=0 (InfiniBand fix: 56ms → 49ms/step)
- 10 layers (up from 9)
- Spectral embedding init (overtone)
- Phase-transition resid_mix
- Muon weight decay 0.02
- FP16 tied embedding export
- Sliding window eval (EVAL_STRIDE=64)
- WARMDOWN_ITERS=3200 (tuned for our step count)
